src/Dashboard/ApiControllers/FunctionsController.cs

```diff
@@ -341,6 +341,7 @@ private InvocationLogViewModel CreateInvocationEntry(RecentInvocationEntry entry
             metadataSnapshot.EndTime = entry.EndTime;
             metadataSnapshot.Succeeded = entry.Succeeded;
             metadataSnapshot.Heartbeat = entry.Heartbeat;
+            metadataSnapshot.HeartbeatExpiredTime = entry.HeartbeatExpiredTime;
```

Comment:
I can try - the challenge is: 
a) they're 2 different heartbeat models (this has the heartbeat on the function; the other has the heartbeat on the host, and the function points to the host, so it's an extra IO lookup) 
b) I had already undergone significant pain trying to play well with existing infrastructure and just ran out of steam ;)